sklearn/linear_model/_theil_sen.py

```diff
@@ -411,6 +415,13 @@ def fit(self, X, y):
         self : returns an instance of self.
             Fitted `TheilSenRegressor` estimator.
         """
+        if self.copy_X != "deprecated":
+            warnings.warn(
+                "The parameter copy_X is deprecated as of version 1.6 and "
+                "will be removed in 1.8.",
```

Comment:
```suggestion
                "`copy_X` was deprecated in 1.6 and will be removed in 1.8 since it "
                "has no effect internally. Simply leave this parameter to its default "
                "value to avoid this warning.",
```

Would be nice to clearly point out a fix in the deprecation message.